File: app/src/main/java/uk/co/sundroid/activity/info/fragments/AboutFragment.java
package uk.co.sundroid.activity.info.fragments;

import android.os.Build;
import android.os.Build.VERSION;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import uk.co.sundroid.AbstractFragment;
import uk.co.sundroid.BuildConfig;
import uk.co.sundroid.R.id;
import uk.co.sundroid.R.layout;

public class AboutFragment extends AbstractFragment {

	@Override
	public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle state) {
		if (container == null) {
			return null;
		}
		View view = inflater.inflate(layout.frag_info_about, container, false);
        String version = String.format("Sundroid %s$1", BuildConfig.VERSION_NAME);
        textInView(view, id.aboutVersion, version);
        textInView(view, id.aboutCodes, getVersionCodes());
		return view;
	}

    public String getVersionCodes() {
        return "Build: " +
                BuildConfig.VERSION_CODE +
                ", API: " +
                VERSION.SDK_INT +
                "\n" +
                Build.MANUFACTURER +
                " " +
                Build.DEVICE;
    }

}

File: app/src/main/kotlin/uk/co/sundroid/activity/info/fragments/AboutFragment.kt
package uk.co.sundroid.activity.info.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import uk.co.sundroid.AbstractFragment
import uk.co.sundroid.BuildConfig

import uk.co.sundroid.activity.MainActivity
import uk.co.sundroid.databinding.FragAboutBinding

class AboutFragment : AbstractFragment() {

    private lateinit var b: FragAboutBinding

    override fun onResume() {
        super.onResume()
        (activity as MainActivity).apply {
            setToolbarTitle("About")
            setToolbarSubtitle(null)
        }
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, state: Bundle?): View? {
        b = FragAboutBinding.inflate(layoutInflater)
        return b.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        val version = String.format("Version %s", BuildConfig.VERSION_NAME)
        b.aboutVersion.text = version
        b.aboutSource.setOnClickListener { browseTo("https://github.com/davemorrissey/sundroid") }
    }
}
